tests/unit/test_args.py

```diff
@@ -459,3 +459,226 @@ def test_client_config_beats_config_store(self):
+        self.args_create._should_set_global_sts_endpoint.assert_called_once_with(
+            region_name='my-region-2',
+            endpoint_url=None,
+            endpoint_config=None,
+        )
+
+    def test_builtin_s3_global_endpoint(self):
+        # The only reason for this builtin to not have the default value
+        # (False) is that the ``_should_force_s3_global`` method
+        # returns True.
+        self.args_create._should_force_s3_global = mock.Mock(return_value=True)
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults()
+        self.assertTrue(bins['AWS::S3::UseGlobalEndpoint'])
+        self.args_create._should_force_s3_global.assert_called_once()
+
+    def test_builtin_s3_accelerate(self):
+        # This builtin has a default value of False. It will be True only if
+        # the ``use_accelerate_endpoint`` field in the S3 configuration is
+        # True.
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            s3_config={'use_accelerate_endpoint': True},
+        )
+        self.assertEqual(bins['AWS::S3::Accelerate'], True)
+
+    def test_builtin_force_path_style(self):
+        # This builtin has a default value of False. It will be True only if
+        # the ``addressing_style`` field in the S3 configuration equals "path".
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            s3_config={'addressing_style': 'path'},
+        )
+        self.assertEqual(bins['AWS::S3::ForcePathStyle'], True)
+
+    def test_builtin_use_arn_region_false(self):
+        # These two builtins both take their value from the ``use_arn_region``
+        # in the S3 configuration, but have different default values.
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            s3_config={'use_arn_region': False},
+        )
+        self.assertEqual(bins['AWS::S3::UseArnRegion'], False)
+        self.assertEqual(bins['AWS::S3Control::UseArnRegion'], False)
+
+    def test_builtin_use_arn_region_true(self):
+        # These two builtins both take their value from the ``use_arn_region``
+        # in the S3 configuration, but have different default values.
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            s3_config={'use_arn_region': True},
+        )
+        self.assertEqual(bins['AWS::S3::UseArnRegion'], True)
+        self.assertEqual(bins['AWS::S3Control::UseArnRegion'], True)
+
+    def test_builtin_disable_mrap(self):
+        # This builtin has a default value of False. It will be True only if
+        # the ``s3_disable_multiregion_access_points`` field in the S3 config
+        # is True.
+        bins = self.call_compute_endpoint_resolver_v2_builtin_defaults(
+            s3_config={'s3_disable_multiregion_access_points': True},
+        )
+        self.assertEqual(bins['AWS::S3::DisableMultiRegionAccessPoints'], True)
+
+    def test_builtin_sdk_endpoint_1(self):
```

Comment:
Instead of using 1,2,3 can we make these test names a bit clearer?

e.g.
```
test_sdk_endpoint_both_inputs_set
test_sdk_endpoint_legacy_set_with_builtin_data
test_sdk_endpoint_legacy_set_without_builtin_data
```

Reply:
I'm also just now realizing that builtin is overloaded here. We might want to change the method name for detecting default endpoints.json data. `uses_default_endpoint_data`?

Reply:
Tests are renamed in 87be3a7.

Re overloaded use of term "builtin": Unfortunately, both uses already have precedent. "Builtin data files" is consistent with `botocore.loaders. BUILTIN_DATA_PATH` and `botocore.loaders.BUILTIN_EXTRAS_TYPES`. (And of course the other use of "builtin" comes from the cross-SDK endpoints spec.)